src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -107,10 +239,9 @@ internal static TupleTypeSymbol ConstructTupleTypeSymbol(TupleTypeSymbol origina
         /// </summary>
         private static WellKnownType GetTupleType(int arity)
         {
-            if (arity > 7)
+            if (arity > 8)
             {
-                // PROTOTYPE
-                arity = 1;
+                throw ExceptionUtilities.Unreachable;
             }
```

Comment:
Is the ```if (arity > 8) throw Unreachable;``` necessary? Indexing into `tupleTypes` will already throw if out of range. If it's necessary consider using ```if (arity > tupleTypes.Length)``` instead. Same comment for `GetTupleCtor` below.